pkg/catalogv2/system/system.go

```diff
@@ -122,27 +122,26 @@ func (m *Manager) installCharts(charts map[desiredKey]map[string]interface{}) {
 	}
 }
 
-func (m *Manager) Uninstall(namespace, name string) error {
-	helmcfg := &action.Configuration{}
-	if err := helmcfg.Init(m.restClientGetter, namespace, "", logrus.Infof); err != nil {
+func (m *Manager) Uninstall(namespace, name string, wait bool) error {
+	uninstall, err := json.Marshal(types.ChartUninstallAction{
+		Timeout: &metav1.Duration{Duration: 5 * time.Minute},
+	})
+	if err != nil {
 		return err
 	}
 
-	l := action.NewList(helmcfg)
-	l.Filter = "^" + name + "$"
-
-	releases, err := l.Run()
+	op, err := m.operation.Uninstall(m.ctx, installUser, namespace, name, bytes.NewBuffer(uninstall))
 	if err != nil {
 		return err
 	}
 
-	if len(releases) == 0 {
-		return nil
+	if wait {
+		if err = m.waitPodDone(op); err != nil {
+			return err
+		}
 	}
 
-	uninstall := action.NewUninstall(helmcfg)
```

Comment:
Does it mean that when using this library to install, it doesn't clean up namespaced items? I am not sure if there is a difference between using library or using pod to uninstall.

Reply:
If the objects in the helm release had the namespace in their metadata, then it would clean them up.

However, if the objects did not have a namespace in their metadata and were installed in a non-default namespace, then objects were not getting cleaned up.

There is a way to use the library, but this seemed cleaner. If you think the library should continued to be used, then I can update this PR with that approach instead.

Reply:
Ok fine by either way. Using library might be fast since it doesn't need to involve creating pods, but it is just one time uninstall so it shouldn't matter that much.

Reply:
@thedadams Actually I think we need to check if errors.isnotfound(). I am getting 
```
error syncing 'rancher-charts': handler bootstrap-charts: apps.catalog.cattle.io "rancher-operator" not found, requeuing
```
when trying to use your method. I happen to test something else with your PR.

Reply:
@StrongMonkey Added check for "not found" error.